frontend_tests/node_tests/composebox_typeahead.js

```diff
@@ -1302,6 +1303,16 @@ run_test('begins_typeahead', () => {
     sweden_topics_to_show.push('totally new topic');
     assert_typeahead_equals("#**Sweden>totally new topic", sweden_topics_to_show);
 
+    // time_jump
+    assert_typeahead_equals("!tim", false);
+    assert_typeahead_equals("!timerandom", false);
+    assert_typeahead_equals("!time", ['translated: Pick a date/time']);
+    assert_typeahead_equals("!time(", ['translated: Pick a date/time']);
+    assert_typeahead_equals("!time(something", ['translated: Pick a date/time']);
+    assert_typeahead_equals("!time(something", ") ", ['translated: Pick a date/time']);
+    assert_typeahead_equals("!time(something)", false);
+    assert_typeahead_equals("!time(something) ", false); // Already completed a topic.
```

Comment:
I feel like we need some sort of unit test for the interaction with the date picker (i.e. if the date picker function returns `123`, how does that edit the compose box string?).  I think the function extractions I suggested would allow the mocking you'd need to do this.